pkg/controllers/managementuser/resourcequota/resource_quota_sync.go

```diff
@@ -304,72 +303,81 @@ func (c *SyncController) createDefaultLimitRange(ns *corev1.Namespace, spec *cor
 	return err
 }
 
-func (c *SyncController) validateAndSetNamespaceQuota(ns *corev1.Namespace, quotaToUpdate string) (bool, *corev1.Namespace, error) {
+func (c *SyncController) validateAndSetNamespaceQuota(ns *corev1.Namespace, quotaToUpdate *v32.NamespaceResourceQuota) (bool, *corev1.Namespace, corev1.ResourceList, error) {
 	if ns == nil || ns.DeletionTimestamp != nil {
-		return true, ns, nil
+		return true, ns, nil, nil
 	}
 
 	// get project limit
 	projectLimit, projectID, err := getProjectResourceQuotaLimit(ns, c.ProjectLister)
 	if err != nil {
-		return false, ns, err
+		return false, ns, nil, err
 	}
 
 	if projectLimit == nil {
-		return true, ns, err
+		return true, ns, nil, err
 	}
 
 	updatedNs := ns.DeepCopy()
-	if quotaToUpdate != "" {
+	if quotaToUpdate != nil {
 		if updatedNs.Annotations == nil {
 			updatedNs.Annotations = map[string]string{}
 		}
-		updatedNs.Annotations[resourceQuotaAnnotation] = quotaToUpdate
+		b, err := json.Marshal(quotaToUpdate)
+		if err != nil {
+			return false, nil, nil, err
+		}
+		updatedNs.Annotations[resourceQuotaAnnotation] = string(b)
 		updatedNs, err = c.Namespaces.Update(updatedNs)
 		if err != nil {
-			return false, updatedNs, err
+			return false, updatedNs, nil, err
 		}
 	}
 
 	// validate resource quota
 	mu := validate.GetProjectLock(projectID)
 	mu.Lock()
 	defer mu.Unlock()
-	// get other Namespaces
+
+	// Get other namespaces' limits.
+	nsLimits, err := c.getNamespacesLimits(ns, projectID)
+	if err != nil {
+		return false, nil, nil, err
```

Comment:
Clarification Question:

Why do you return Nil for the namespace here but return `updatedNs` along with the error in L349?

Reply:
That's a good catch. The original code returned `updatedNs` in this case.